phoenix-pherf/src/main/java/org/apache/phoenix/pherf/util/ResultsComparator.java

```diff
@@ -0,0 +1,353 @@
+		this.setLabelsAsString(labels);
+	}
+	
+	String[] getLabels() {
+		return labels;
+	}
+
+	void setLabels(String[] labels) {
+		this.labels = labels;
+	}
+
+	void setLabelsAsString(String labels) {
+		this.labels = labels.split(",");
+	}
+	
+	public void readAndRender() {
+		try {
+			for (String label : labels) {
+				read(label);
+			}
+			renderAsGoogleChartsHTML();
+
+		} catch (Exception e) {
+			e.printStackTrace();
+		}
+	}
+
+	/**
+	 * Reads aggregate file and convert it to DataNode 
+	 * @param label
+	 * @throws Exception
+	 */
+    private void read(String label) throws Exception {
+		String resultFileName = resultDir 
+				+ PherfConstants.PATH_SEPARATOR
+				+ PherfConstants.RESULT_PREFIX 
+				+ label
+				+ ResultFileDetails.CSV_AGGREGATE_PERFORMANCE.getExtension();
+
+    	FileReader in = new FileReader(resultFileName);
+    	final CSVParser parser = new CSVParser(in, CSVFormat.DEFAULT.withHeader());
+
+        for (CSVRecord record : parser) {
+            String group = record.get("QUERY_GROUP");
+            String query = record.get("QUERY");
+            String explain = record.get("EXPLAIN_PLAN");
+            String tenantId = record.get("TENANT_ID");
+            long avgTime = Long.parseLong(record.get("AVG_TIME_MS"));
+            long minTime = Long.parseLong(record.get("AVG_MIN_TIME_MS"));
+            long numRuns = Long.parseLong(record.get("RUN_COUNT"));
+            long rowCount = Long.parseLong(record.get("RESULT_ROW_COUNT"));
+            Node node = new Node(minTime, avgTime, numRuns, explain, query, tenantId, label, rowCount);
+            
+            if (datanodes.containsKey(group)) {
+            	datanodes.get(group).getDataSet().put(label, node);
+            } else {
+            	datanodes.put(group, new DataNode(label, node));
+            }
+        }
+        parser.close();
+    }
+
+    /**
+     * Verifies if the first result is within the set 
+     * threshold of pherf.default.comparison.threshold
+     * set in pherf.properties files
+     * @param threshold
+     * @return
+     */
+    private boolean verifyWithinThreshold(double threshold) {
+    	long resetTimeToCompare = -1;
+    	long timeToCompare = resetTimeToCompare;
+    	for (Map.Entry<String, DataNode> dn : datanodes.entrySet()) {    	   	
+        	for (Map.Entry<String, Node> node : dn.getValue().getDataSet().entrySet()) {
+        		if (timeToCompare == -1) {
+        			timeToCompare = node.getValue().getMinTime();
+        		}
+				if ((((double) (timeToCompare - node.getValue().getMinTime())) / (double) node
+						.getValue().getMinTime()) > threshold) {
+					return false;
+				}
+        	}
+        	timeToCompare = resetTimeToCompare;
+    	}
+    	return true;
+    }
+    
+    /**
+     * Render results as Google charts
+     * @throws FileNotFoundException
+     * @throws UnsupportedEncodingException
+     */
+    private void renderAsGoogleChartsHTML() throws FileNotFoundException, UnsupportedEncodingException {
```

Comment:
Should all this chart stuff go in a util object? It seems unrelated to a Result comparison.